src/main/java/com/r307/arbitrader/service/TradingService.java

```diff
@@ -568,9 +615,62 @@ private void logEntryTrade(Spread spread, String shortExchangeName, String longE
 
     }
 
-    // get volume for an entry position considering exposure and exchange step size if there is one
-    private BigDecimal getVolumeForEntryPosition(BigDecimal maxExposure, BigDecimal price, int scale) {
-        return maxExposure.divide(price, scale, RoundingMode.HALF_EVEN);
+    /**
+     * Calculates the volume to trade on the long exchange such as:
+     * - the total price does not exceed the long exchange maximum exposure
+     * - the total price to trade on the short exchange does not exceed the short exchange maximum exposure
+     * @see #getShortVolumeFromLong and #getFeeFactor
+     * Detailed maths: https://github.com/scionaltera/arbitrader/issues/325
+     */
+    private BigDecimal getVolumeForLongTrade(BigDecimal longMaxExposure, BigDecimal shortMaxExposure, BigDecimal longPrice, BigDecimal shortPrice, BigDecimal longFee, BigDecimal shortFee) {
+        //volume limit induced by the maximum exposure on the long exchange
+        BigDecimal longVolume1 = longMaxExposure.divide(longPrice,RoundingMode.HALF_EVEN);
+
+        //volume limit induced by the maximum exposure on the short exchange: shortVolume * shortPrice == shortMaxExposure
+        //to respect market neutrality: shortVolume = longVolume2 / feeFactor
+        BigDecimal longVolume2 = getFeeFactor(longFee,shortFee).multiply(shortMaxExposure).divide(shortPrice,RoundingMode.HALF_EVEN);
+        return longVolume1.min(longVolume2);
+    }
+
+    /**
+     * Calculates a market neutral volume to trade on the short exchange from the volume to trade on the long exchange
+     * @see #getVolumeForLongTrade and #getFeeFactor
+     */
+    private BigDecimal getShortVolumeFromLong(BigDecimal longVolume, BigDecimal longFee, BigDecimal shortFee) {
```

Comment:
I like how you broke these up. It would make it pretty easy to make them package private and write unit tests to guarantee they're doing the math correctly. ;)

Reply:
I will move the methods to a TradeVolumes class, and each public method of the class could be unit tested to make sure our volume related maths is correct.